src/solvers/dgsem_p4est/dg_2d_parabolic.jl

```diff
@@ -454,22 +456,137 @@ function prolong2boundaries!(cache_parabolic, flux_viscous,
         j_node = j_node_start
         for i in eachnode(dg)
             # this is the outward normal direction on the primary element
-            normal_direction = get_normal_direction(primary_direction,
-                                                    contravariant_vectors,
-                                                    i_node, j_node, primary_element)
+            normal_direction = get_normal_direction(direction, contravariant_vectors,
+                                                    i_node, j_node, element)
 
             for v in eachvariable(equations_parabolic)
-                flux_viscous = SVector(flux_viscous_x[v, i_primary, j_primary,
-                                                      primary_element],
-                                       flux_viscous_y[v, i_primary, j_primary,
-                                                      primary_element])
+                flux_viscous = SVector(flux_viscous_x[v, i_node, j_node, element],
+                                       flux_viscous_y[v, i_node, j_node, element])
 
                 boundaries.u[v, i, boundary] = dot(flux_viscous, normal_direction)
             end
             i_node += i_node_step
```

Comment:
where is the i_node initially set? I understand that this interface routine needs to account for flipped index directions but the loop runs over `i` where as the data access uses `i_node`. But I cannot see where this `i_node` index is ever updated.

Reply:
Isn't it in the last line in the code snipped?

Reply:
It's set in https://github.com/trixi-framework/Trixi.jl/blob/3796dc8535f35adbf5314e6744dd241d1aac4608/src/solvers/dgsem_p4est/dg_2d_parabolic.jl#L455 and updated in https://github.com/trixi-framework/Trixi.jl/blob/3796dc8535f35adbf5314e6744dd241d1aac4608/src/solvers/dgsem_p4est/dg_2d_parabolic.jl#L468